examples/siro_sandbox/gui_navigation_helper.py

```diff
@@ -100,11 +95,85 @@ def viz_and_get_humanoid_walk_dir(self):
             path_points, path_endpoint_radius, path_color
         )
 
-        if len(path.points) >= 2:
-            walk_dir = mn.Vector3(path.points[1]) - mn.Vector3(path.points[0])
-            return walk_dir
+    def get_humanoid_walk_hints_from_remote_gui_input(
+        self, visualize_path=True
+    ):
+        walk_dir = None
+        distance_multiplier = 1.0
+
+        (
+            target_pos,
+            target_rot_quat,
+        ) = self._sandbox_service.remote_gui_input.get_head_pose()
+
+        walk_dir, distance_multiplier = self._get_humanoid_walk_hints(
+            target_pos=target_pos,
+            target_rot_quat=target_rot_quat,
+            visualize_path=visualize_path,
+        )
+
+        return walk_dir, distance_multiplier
+
+    def get_humanoid_walk_hints_from_ray_cast(self, visualize_path=True):
+        walk_dir = None
+        distance_multiplier = 1.0
+
+        target_on_floor = self._get_target_pos_from_ray_cast()
+        if target_on_floor is None:
+            return walk_dir, distance_multiplier
+
+        walk_dir, distance_multiplier = self._get_humanoid_walk_hints(
+            target_pos=target_on_floor,
+            target_rot_quat=habitat_sim.utils.common.random_quaternion(),  # habitat_sim.utils.common.random_quaternion() can be used to generate random rotations for testing
+            visualize_path=visualize_path,
+        )
+
+        return walk_dir, distance_multiplier
+
+    def _get_humanoid_walk_hints(
+        self, target_pos, target_rot_quat, visualize_path=True
+    ):
+        walk_dir = None
+        distance_multiplier = 1.0
+        geodesic_dist_threshold = 0.05
+
+        found_path, path = self._get_humanoid_walk_path_to(target_pos)
+        if (
+            found_path
+            and len(path.points) >= 2
+            and path.geodesic_distance >= geodesic_dist_threshold
+        ):
+            walk_dir = self._get_humanoid_walk_dir_from_path(path)
+            distance_multiplier = 1.0
+            if visualize_path:
+                self._viz_humanoid_walk_path(path)
+
+        if walk_dir is None and target_rot_quat is not None:
+            walk_dir = self._compute_forward_dir(target_rot_quat)
+            distance_multiplier = 0.0
+
+        return walk_dir, distance_multiplier
+
+    def _get_target_pos_from_ray_cast(self):
+        ray = self._sandbox_service.gui_input.mouse_ray
+
+        floor_y = 0.15  # hardcoded to ReplicaCAD
+
+        if not ray or ray.direction.y >= 0 or ray.origin.y <= floor_y:
+            return None
+
+        dist_to_floor_y = (ray.origin.y - floor_y) / -ray.direction.y
+        target_on_floor = ray.origin + ray.direction * dist_to_floor_y
+
+        return target_on_floor
+
+    def _compute_forward_dir(self, target_rot_quat):
```

Comment:
Used [this code](https://github.com/facebookresearch/habitat-lab/blob/929c0efea33a34a786e75430735c3da9f7bff233/habitat-lab/habitat/articulated_agent_controllers/humanoid_rearrange_controller.py#L129) as an example.